packages/typescript-estree/src/convert.ts

```diff
@@ -3458,4 +3459,20 @@ export class Converter {
 
     throw createError(message, this.ast, start, end);
   }
+  #checkForStatementDeclaration(initializer: ts.ForInitializer): void {
+    if (ts.isVariableDeclarationList(initializer)) {
+      if (
+        !(
+          initializer.flags & ts.NodeFlags.Const ||
+          initializer.flags & ts.NodeFlags.Let ||
+          initializer.flags === ts.NodeFlags.None
+        )
+      ) {
+        this.#throwError(
+          initializer,
+          " The left-hand side of a 'for...in' statement cannot be a 'using' declaration.",
```

Comment:
nit remove the space

```suggestion
          "The left-hand side of a 'for...in' statement cannot be a 'using' declaration.",
```